internal/controllers/core/liveupdate/reconciler.go

```diff
@@ -925,63 +925,87 @@ func (r *Reconciler) applyInternal(
 		}
 	}
 
-	var lastExecErrorStatus *v1alpha1.LiveUpdateContainerStatus
+	copierContext, cancelCopiers := context.WithCancel(ctx)
+	wg := sync.WaitGroup{}
+	results := make(chan *v1alpha1.LiveUpdateStatus)
 	for _, cInfo := range containers {
-		// TODO(nick): We should try to distinguish between cases where the tar writer
-		// fails (which is recoverable) vs when the server-side unpacking
-		// fails (which may not be recoverable).
-		archive := build.TarArchiveForPaths(ctx, toArchive, nil)
-		err = cu.UpdateContainer(ctx, cInfo, archive,
-			build.PathMappingsToContainerPaths(toRemove), boiledSteps, hotReload)
-		_ = archive.Close()
-
-		lastFileTimeSynced := input.LastFileTimeSynced
-		if lastFileTimeSynced.IsZero() {
-			lastFileTimeSynced = apis.NowMicro()
-		}
+		go func(cInfo liveupdates.Container) {
+			wg.Add(1)
```

Comment:
this needs to be outside the goroutine (i think CI has a lint check for this mistake)

though i would probably switch to using an errgroup, as in this example: 
https://pkg.go.dev/golang.org/x/sync/errgroup#example-Group-Parallel
which is a bit easier to reason about

Reply:
Cool, I didn't know about this! This use-case doesn't seem to fit perfectly, but it does clean things up a bit.